tools/modules/system/module_armbian_firmware.sh

```
@@ -198,26 +201,26 @@ function module_armbian_firmware() {
-						packages+="${pkg}=${version} ";
+					if [[ -n $(echo "$cache_show" | grep "$version""$") ]]; then
+						packages+="${pkg}=${version} "
 					fi
 				elif [[ -n "${cache_show}" ]]; then
-					packages+="${pkg} ";
+					packages+="${pkg} "
 				fi
 			done
 
 			# if this is called with a parameter hide, we only prepare this list but don't show its content
 			[[ "$4" != "hide" ]] && echo ${packages[@]}
```

Shellcheck (0.10.0)

[error] Double quote array expansions to avoid re-splitting elements.

(SC2068)